Recommend LSP's in development docs

ryneeverett opened the pull request:
Per conversation in #3338.

There are already a lot of documented compile options so I think we're better off suggesting that everybody create a compile_commands.json whether or not they're using an LSP because it doesn't cost much.

While I was at it it seemed reasonable to mention rust LSP too. Now that rls is deprecated I'm not sure there is any competitor to rust-analyzer worth mentioning.

ryneeverett committed:
Recommend LSP's in development docs

Per conversation in #3338.

There are already a lot of documented compile options so I think we're
better off suggesting that everybody create a compile_commands.json
whether or not they're using an LSP because it doesn't cost much.

While I was at it it seemed reasonable to mention rust LSP too. Now that
rls is deprecated I'm not sure there is any competitor to rust-analyzer
worth mentioning.

felixschurk commented:
Would it maybe then not just make sense to set it in the CMake list, such that everybody gets them by default?
This would keep the compile commands a bit cleaner.

As you said they do not hinder anyone at anything, they only would help.

felixschurk commented:
It would be adding following line after the `cmake_minimum_required ...`

`set(CMAKE_EXPORT_COMPILE_COMMANDS ON)`
like that

```cmake
cmake_minimum_required (VERSION 3.22)
set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 
project (task
...........
```

ryneeverett committed:
Export compile commands by default.

Thanks to @felixschurk for the idea and telling me how to do it.

It took me a minute to figure out that this places the
compile_commands.json in the build directory rather than the root of the
project. But clangd still finds it there and that's a better place for
it anyway.

felixschurk commented:
Ah yes sorry that I did not mention.

Yes, it places it in the `build` directory as one could have multiple build directories with different `compile_command.json` settings e.g. different build systems, compiler or even with additional flags such as coverage etc.

djmitche reviewed (approved):
Nice, thanks! Rust analyzer works well with the TC code out of the box.

djmitche commented:
The rust failures are my fault
